### PR TITLE
fix: handled value as string from queryParams for trace filters

### PR DESCRIPTION
**Pull request by SagarRajput-7**
### Summary

Here the serviceName value coming as a string from queryParam and as spreading and includes works on string also, which was splitting the string into individual chars and indirectly making the value as an array of string with chars in it. Also, this was breaking the page due to missing null check.

#### Screenshots

https://github.com/SigNoz/signoz/assets/162284829/880f2cef-a93e-4b50-a8c8-890f6758339c

### CODE REVIEW

File: frontend/src/pages/TracesExplorer/Filter/SectionContent.tsx [hunk 1]
```diff
@@ -92,17 +92,21 @@ export function SectionBody(props: SectionBodyProps): JSX.Element {
 		if (checked) {
 			addFilter(type, newValue, setSelectedFilters, keys);
 			setCheckedItems((prev) => {
-				if (!prev.includes(newValue)) {
-					prev.push(newValue);
+				const arr = prev || [];
+				if (!arr.includes(newValue)) {
```
Comment (vikrantgupta25) [inline frontend/src/pages/TracesExplorer/Filter/SectionContent.tsx:96]: string can still go inside the if block ?

**Comment by vikrantgupta25** [comment]
>  isArray logic treats the string also as an array

This is not correct. https://lodash.com/docs/4.17.15#isArray

File: frontend/src/pages/TracesExplorer/Filter/SectionContent.tsx (hunk 1 above)
Comment (SagarRajput-7) [reply frontend/src/pages/TracesExplorer/Filter/SectionContent.tsx:96]: done, added isArray check

**Comment by SagarRajput-7** [comment]
> > isArray logic treats the string also as an array
> 
> This is not correct. https://lodash.com/docs/4.17.15#isArray

yeah.
Here - as spreading and includes works on string also, which was splitting the string into individual chars and indirectly making the value as an array of string with chars in it

**Review by vikrantgupta25 — APPROVED** [review]

**Review by YounixM — APPROVED** [review]

**Pull request merged**